### PR TITLE
Game may no longer have multiple players with the same player ID, and `firstPlayer` must be in the list of players.

### PR DESCRIPTION
**Pull request by kberg**
This only happens in tests, and as you can see by this change, it's *a lot* of tests, which casually use `[player, player]` as their list of players in a game. This causes some super-weird test failures when a player receives both ends of an action. We shouldn't be doing it anyway.

There were surprisingly few tests that didn't like this: MiningGuild and GameLoader. Both were easy enough to fix, ... no actually MiningGuild was a pain in the butt. Mining Guild's issue was that that it used maxOutOceans, which by default stops when it places 9 oceans -- but there are 12 ocean spaces! So I switched to a more direct space filter.

GameLoader just needed dedicated Player IDs. Also a bit of a pain trying to manage the sheer volume of callbacks between GameLoader and tests.

**What do you need, reviewer?**

It's a lot of files. Do you want me to split it out? No problem. Almost all the changes are dead simple to understand (except MiningGuild and GameLoader, but I've discussed that.) Also, do you want me to rename `redPlayer`? I am not terribly fond of `player2` - you tell me.

### CODE REVIEW

File: tests/database/GameLoader.spec.ts [hunk 1]
```diff
@@ -71,13 +72,14 @@ describe('GameLoader', function() {
     });
     expect(actual).to.be.undefined;
     loader.start();
+    console.log(actual!.getPlayers());
```
Comment (Lynesth) [inline tests/database/GameLoader.spec.ts:75]: Is this supposed to stay there ?

File: src/Game.ts [hunk 2]
```diff
@@ -186,8 +186,17 @@ export class Game implements ISerializable<SerializedGame> {
       public id: GameId,
       private players: Array<Player>,
       private first: Player,
-      // ... creates a shallow copy.
       public gameOptions: GameOptions = {...DEFAULT_GAME_OPTIONS}) {
+      {
+        const _playerIds = players.map((p) => p.id);
+        if (_playerIds.find((pid) => pid === first.id) === undefined) {
```
Comment (Lynesth) [inline src/Game.ts:192]:
```ts
        if (_playerIds.includes(first.id) === false) {
```

**Review by Lynesth — APPROVED** [review]

File: tests/database/GameLoader.spec.ts (hunk 1 above)
Comment (kberg) [reply tests/database/GameLoader.spec.ts:75]: Nope! Sorry. Thanks.

File: src/Game.ts (hunk 2 above)
Comment (kberg) [reply src/Game.ts]: Yesss... thank you.

**Pull request merged**